tw-tkms-starter/src/main/java/com/transferwise/kafka/tkms/config/TkmsConfiguration.java

```diff
@@ -66,14 +60,9 @@ public TransactionalKafkaMessageSender tkmsTransactionalKafkaMessageSender() {
   }
 
   @Bean
-  @ConditionalOnMissingBean(ITkmsDao.class)
-  public TkmsDao tkmsDao(ITkmsDataSourceProvider dataSourceProvider, TkmsProperties tkmsProperties,
-      ITkmsMetricsTemplate metricsTemplate, ITkmsMessageSerializer messageSerializer,
-      ITransactionsHelper transactionsHelper, IProblemNotifier problemNotifier) {
-    if (tkmsProperties.getDatabaseDialect() == DatabaseDialect.POSTGRES) {
-      return new TkmsPostgresDao(dataSourceProvider, tkmsProperties, metricsTemplate, messageSerializer, transactionsHelper, problemNotifier);
-    }
-    return new TkmsMariaDao(dataSourceProvider, tkmsProperties, metricsTemplate, messageSerializer, transactionsHelper, problemNotifier);
+  @ConditionalOnMissingBean(ITkmsDaoProvider.class)
+  public TkmsDaoProvider tkmsDaoProvider() {
```

Comment:
Yep, in that case Spring can find all interfaces of that exact type. In case a bean implements multiple interfaces, each of which will be injected into different places.

It is not the case here, but it is a good practice. Has helped me to avoid quite a many anomalies investigations.